lua/tabnine/chat.lua

```diff
@@ -0,0 +1,38 @@
+local chat_binary = require("tabnine.chat_binary")
+local fn = vim.fn
+local utils = require("tabnine.utils")
+local tabnine_binary = require("tabnine.binary")
+local api = vim.api
+
+local M = {}
+
+local function register_events()
+	chat_binary:register_event("init", function(_, answer)
+		answer({ ide = "ij" })
+	end)
+
+	chat_binary:register_event("get_chat_state", function(_, answer)
+		answer({ conversations = {} })
+	end)
+	chat_binary:register_event("get_user", function(_, answer)
+		tabnine_binary:request({ State = { dummy = true } }, function(state)
+			answer({ token = state.access_token, username = state.user_name })
+		end)
+	end)
+	chat_binary:register_event("insert-at-cursor", function(message, _)
+		local lines = utils.str_to_lines(message.code)
+		api.nvim_buf_set_text(0, fn.line(".") - 1, fn.col(".") - 1, fn.line(".") - 1, fn.col(".") - 1, lines)
```

Comment:
Can this overwrite the text in the buffer? I'm not familiar with `nvim_buf_set_text`

Reply:
Yep. Overwriting/adding text. What are you using for inserting text?

Reply:
I have no experience whatsoever *correctly* inserting text, I've used nvim_feedkeys any time I needed to (bad idea). Just wanted to make sure we weren't *unintentionally* deleting text currently in the buffer.